Add codesign utility function

gitperr opened the pull request:
Calls on `rcodesign` to code sign a target binary.

TODO:
- [x] ~Add tests~ (no good way of doing this, `rcodesign verify` does not work very well with this warning: ```(the verify command is known to be buggy and gives misleading results; we highly recommend using Apple's tooling until this message is removed)```
- [x] Other, based on feedback

gitperr commented:
@SuperQ wanna take a look?

SuperQ commented on cmd/codesign.go:
```diff
@@ -0,0 +1,53 @@
+// Copyright © 2016 Prometheus Team
```
New file needs current year.
```suggestion
// Copyright © 2024 Prometheus Team
```

gitperr replied:
Updated to 2024

SuperQ reviewed (changes requested):
Minor nit, otherwise LGTM.

We're also going to need to plumb this into our CI templates so that it's done automatically.

gitperr commented:
Squashed, will make changes to https://github.com/prometheus/node_exporter/pull/2916 as well to accompany this.

gitperr commented:
Updated README as well now. And squashed again. Should be good to go, please take another look.

This should be merged before https://github.com/prometheus/node_exporter/pull/2916 to make pipelines start working properly with `promu codesign`

SuperQ reviewed (approved):
LGTM

gitperr commented:
@roidelapluie does this PR look good?

gitperr commented:
Is there anyone else we can ask review from? If roidelapluie is busy, maybe someone else could take a look?
